### DISCUSSION TITLE
Error: heap limit reached

### DISCUSSION THREAD

**Discussion by werkstattcodes** [quarto-dev/quarto-cli#3344] [Q&A] [answered]
Hello,

I am working on a quarto document (VS Code) which includes a reactable table (r package) with relative verbose text content (around 80000 rows, each containing a paragraph of text, the file underlying the table has around 100 MB as a csv). I am able to render the table when feeding in only a sample of rows, or when rendering only the full table in another qmd (without other parts of my main document).

However, when rendering the entire document, the process at one point is aborted with the error message

```
  |......................................................................| 100%
  ordinary text without R code


output file: scr_analysis.knit.md

There were 12 warnings (use warnings() to see them)

<--- Last few GCs --->

[19168:0000023F0794AD00]   222310 ms: Mark-sweep 1382.8 (1426.1) -> 1382.6 (1425.9) MB, 7.7 / 0.0 ms  (average mu = 0.999, current mu = 0.938) allocation failure; scavenge might not 
succeed
[19168:0000023F0794AD00]   222328 ms: Mark-sweep (reduce) 1398.2 (1441.4) -> 1398.2 (1409.9) MB, 10.7 / 0.0 ms  (+ 0.0 ms in 0 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 11 ms) (average mu = 0.998, current mu = 0.39

<--- JS stacktrace --->


#
# Fatal javascript OOM in Reached heap limit
#
```

When observing the file folder in which the qmd script is located, I can see that for a brief moment a 'knit.md' of approx 103 MB in size is produced. However, it exists only briefly, there is no conversion to a html file.

I don't know whether this problem is related to reactable or quarto, but I hope there is some way to get the document incl. the full table working. 

Maybe it's related to the V8 library, and probably this [issue](https://github.com/jeroen/V8/issues/33). 

Any idea how to overcome this problem? I already filed an [issue ](https://github.com/glin/reactable/issues/287)at the reactable repository, but maybe there is some quarto memory feature which is relevant?

Many thanks!

**Comment by ethantenison** [comment]
Just had the same issues x_x

**Reply by ShanSabri** [reply]
I also have the same issue!

**Comment by mcanouil** [comment]
Beside the fact that that having such big document really seems like a bad idea, the issue is actually upstream and from Pandoc, see <https://duckduckgo.com/?q=pandoc+heap+memory>.

Note that you can pass it arguments to Pandoc using Quarto CLI.

**Reply by cscheid** [reply]
@mcanouil That's actually incorrect for this report:

```
#
# Fatal javascript OOM in Reached heap limit
#
```

Pandoc is written in Haskell :)

The problem has been reported by other users in the linked issue; there's not much we can do besides recommending users to experiment with v8 heap options, as described there.

**Reply by mcanouil** [reply]
@cscheid Thanks for setting me straight on this. I am so used to get the heap limit using Pandoc itself in the past that I did not noticed the clearly written "JavaScript" part 🤦‍♂️ (also maybe 1am is not the best time to think clearly ^^)

**Comment by philipgladwin** [comment]
Is there any update on this?  I am in the same boat as others who are trying to render but unable to progress because of this memory issue.  Thank you,

**Comment by mcanouil** [comment, marked as answer]
Closing this in favour of #2462
